### PR TITLE
Eliminate all print, println, eprintln calls from the codebase

### LINKED ISSUE #158: Eliminate all `print!` and `println!` calls from the codebase
Now that we're using [tracing](https://docs.rs/tracing/latest/tracing/), SlateDB should not be printing anywhere. The codebase still has a lot of print statements left over from before. These need to get converted to tracing log calls.

### PR DESCRIPTION
**Pull request by mohnishbasha**
Issue:
- https://github.com/slatedb/slatedb/issues/158

Fix:
- [x] Did a generic search across the code based for print, println, eprintln statements to remove it. 
- [x] Replace the print statements with [tracing](https://docs.rs/tracing/latest/tracing/) lib instrumentation.

Tests:
- [x] build and compile
- [x] run unit tests

Fixes #158

### CODE REVIEW

**Comment by criccomini** [comment]
Rather than deleting the statements, can you convert them to log calls using the [`tracing`](https://docs.rs/tracing/latest/tracing/) library that we've added to our Cargo.toml?

**Comment by mohnishbasha** [comment]
> Rather than deleting the statements, can you convert them to log calls using the [`tracing`](https://docs.rs/tracing/latest/tracing/) library that we've added to our Cargo.toml?

yup, that what i was thinking. I will take a look at the tracing lib and replace it with commented code.

File: src/db_bench/main.rs [hunk 1]
```diff
@@ -53,6 +56,18 @@ fn load_object_store(args: &DbBenchArgs) -> Result<Arc<dyn ObjectStore>, SlateDB
 
 #[tokio::main]
 async fn main() {
+
+    // Initialize the tracing subscriber
+    let subscriber = FmtSubscriber::builder()
+        .with_max_level(Level::TRACE)
```
Comment (criccomini) [inline src/db_bench/main.rs:62]: For benching, I think we probably want info-level default. With that in mind, perhaps this is easier:

```rust
tracing_subscriber::fmt::init();
```

File: src/compactor_executor.rs [hunk 2]
```diff
@@ -175,7 +177,7 @@ impl TokioCompactionExecutorInner {
             for result in results {
                 match result {
                     Err(e) if !e.is_cancelled() => {
-                        eprintln!("Shutdown error in compaction task: {:?}", e);
+                         error!("Shutdown error in compaction task: {:?}", e);
```
Comment (criccomini) [inline src/compactor_executor.rs:180]: Nit: extra space before error

**Review by criccomini — COMMENTED** [review]
Looks good. Tests and linting are failing. Please fix.

**Review by agavra — APPROVED** [review]
LGTM

**Review by palash25 — APPROVED** [review]

**Pull request merged**